pkg/apis/serving/v1beta1/inference_service_defaults.go

```diff
@@ -86,4 +89,224 @@ func (isvc *InferenceService) DefaultInferenceService(config *InferenceServicesC
 			}
 		}
 	}
+	isvc.setPredictorModelDefaults()
+}
+
+func (isvc *InferenceService) setPredictorModelDefaults() {
+	if isvc.Spec.Predictor.Model != nil {
```

Comment:
We need to consider when a PATCH or update to an existing inference service happens. For example, if a user deployed an Isvc using the  `predictor.tensorflow` schema, the mutator will convert this to `predictor.model`. Then when a user updates the isvc also using the `predictor.tensorflow` schema (for example, when doing a canary rollout), at this point in the code, the isvc spec will have both `Predictor.Tensorflow`and `Predictor.Model` as non-nil. 

This ties in to the failing canary rollout e2e test where it is failing the `validateExactlyOneImplementation` call above with `Exactly one of [SKLearn, XGBoost, Tensorflow, PyTorch, Triton, ONNX, PMML, LightGBM, Paddle, Model, PodSpec] must be specified in PredictorSpec`

Reply:
@yuzisun your thoughts on how we can handle update/patch request?

Reply:
I think removing  the `isvc.Spec.Predictor.Model != nil` check here, and moving the call to `isvc.setPredictorModelDefaults()` in `DefaultInferenceService` to the beginning of that function should work.